### DISCUSSION TITLE
feature request -- can brokenaxes be modified to act as a wrapper to existing plots?

### DISCUSSION THREAD

**Discussion by PhillipMaire** [bendichter/brokenaxes#80] [Q&A]
Hey thanks for your work on brokenaxes! I was wondering if it is possible to make a wrapper for only basic operations that would work like this. 
```
fig, ax = plt.subplots()
ax.plot(range(10), range(10))
baxes_wrapper = brokenaxes_wrapper(xlims=((0,2),(4,7)), ylims=((0,2),(4, 7)))
fig, ax = baxes_wrapper(fig, ax) # adds break to plot
```
the reason for this is that I plot using pandas dataframe and pivot tables and groupby functions which makes it hard to use brokenaxes as the plotter directly. thank you!

**Comment by bendichter** [comment]
@PhillipMaire thanks for the question. I did try to do something like that a few years ago, but it would require essentially copying all of the plot components to a new figure and axes, and I hit some roadblocks. In theory you should be able to do something like:

```python
import pandas as pd
from brokenaxes import brokenaxes

df = pd.util.testing.makeMixedDataFrame()

bax = brokenaxes(xlims=((1,2), (3,4)))
for ax in bax.axs:
    df.plot(ax=ax)
```

but I just tried that and got a strange bug about exceeding the maximum ticks.

**Reply by PhillipMaire** [reply]
could [this post](https://stackoverflow.com/questions/45810557/pyplot-copy-an-axes-content-and-show-it-in-a-new-figure) solve the problem? I ran that code snippet and obviously ran into the same issue as you did. could we loop through the attributes of `ax` like 
```
fmt_ydata                            type->   NoneType                                         > None
fmt_xdata                            type->   NoneType                                         > None
clipbox                              type->   NoneType                                         > None
ArtistList                           type->   abc.ABCMeta                                      ...lotlib.axes._base._AxesBase.ArtistList'>
use_sticky_edges                     type->   bool                                             > True
stale                                type->   bool                                             > True
mouseover                            type->   bool                                             > False
ignore_existing_data_limits          type->   bool                                             > False
axison                               type->   bool                                             > True
stale_callback                       type->   function                                         ...stale_figure_callback at 0x7f02cb0e84d0>
numRows                              type->   int                                              > 1
numCols                              type->   int                                              > 2
zorder                               type->   int                                              > 0
containers                           type->   list                                             > []
child_axes                           type->   list                                             > []
sticky_edges                         type->   matplotlib.artist._XYPair                        > _XYPair(x=[], y=[])
texts                                type->   matplotlib.axes._base._AxesBase.ArtistList       > <Axes.ArtistList of 0 texts>
patches                              type->   matplotlib.axes._base._AxesBase.ArtistList       > <Axes.ArtistList of 0 patches>
collections                          type->   matplotlib.axes._base._AxesBase.ArtistList       > <Axes.ArtistList of 0 collections>
```
Would you know the necessary attributes? I would just be guessing.

**Reply by PhillipMaire** [reply]
interesting it has no problem when passing index 1, only issues are with index 0 ???? :confused: 
Edited to make more sense 
```
import pandas as pd
from brokenaxes import brokenaxes
df = pd.util.testing.makeMixedDataFrame()
bax = brokenaxes(xlims=((1,2), (3,4)))
df.plot(ax=bax.axs[1])
```
<img width="345" alt="image" src="https://user-images.githubusercontent.com/19584564/160217409-5485b454-2281-4e9a-921f-f4356b586158.png">

**Reply by PhillipMaire** [reply]
also y seems to work fine?!?! Just the first axis of the x axis do you think there may be something different in your code that you may have done to the first axis of the x?? this could potentially be all that is wrong
```
import pandas as pd
from brokenaxes import brokenaxes
df = pd.util.testing.makeMixedDataFrame()
bax = brokenaxes(ylims=((1,2), (3,4)))
df.plot(ax=bax.axs[0])
df.plot(ax=bax.axs[1])
```
<img width="360" alt="image" src="https://user-images.githubusercontent.com/19584564/160217637-f53e5136-dfcc-4bfe-82f2-6d2dce0c63f3.png">

**Reply by bendichter** [reply]
Good job with the investigation so far! Keep going and you'll find the issue.

**Reply by PhillipMaire** [reply]
I have looked through the code and I cant find any reason for this using the method above the y axis version is fully functional. I have compared the attributes of the working `df.plot(ax=bax.axs[1])` and broken axes `df.plot(ax=bax.axs[0])` and using a text compare I can't find anything different other than what is expected to be different (e.g. the actual numbers for the axes etc).  I can only find this work around by making 3 spits and only plotting the 2nd and the 3rd one. there might be a way to remove the other parts of the first subplot but I'm not entirely motivated to do so :sleepy:

```
import pandas as pd
from brokenaxes import brokenaxes
df = pd.util.testing.makeMixedDataFrame()
bax = brokenaxes(xlims=((0, 0),(0,2), (3,4)))

fig = plt.figure()
df.plot(ax=bax.axs[1])
df.plot(ax=bax.axs[2])
bax.axs[0].set_texts = []
```
![image](https://user-images.githubusercontent.com/19584564/160221002-be854f2c-f8c2-4c8d-bb3d-cbdad242dc18.png)

**Reply by bendichter** [reply]
Did you try this approach with your own data? You may not run into this weird bug.
